internal/common/key_codes.rs

```diff
@@ -18,6 +18,15 @@ macro_rules! for_each_special_keys {
 '\u{0019}'  # Backtab     # Qt_Key_Key_Backtab      #              ;
 '\u{007f}'  # Delete      # Qt_Key_Key_Delete       # Delete       ;
 
+// The modifier key codes comes from https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode.
+'\u{0010}'  # Shift       # Qt_Key_Key_Shift        # LShift       ;
+'\u{0010}'  # ShiftR      #                         # RShift       ;
+'\u{0011}'  # Control     # Qt_Key_Key_Control      # LControl     ;
+'\u{0011}'  # ControlR    #                         # RControl     ;
```

Comment:
I don't think that `Shift` and `ShiftR` should share the same code, as well as `Control` and `ControlR` - how else can we distinguish them later?

Reply:
That's from the Mozilla source there are the same keys for left and right. But I think we can use a different one to distinguish between left and right.

Reply:
done